src/edit_view.rs

```diff
@@ -59,20 +82,142 @@ const TOP_PAD: f32 = 6.0;
 const LEFT_PAD: f32 = 6.0;
 const LINE_SPACE: f32 = 17.0;
 
+impl Widget for EditView {
+    fn paint(&mut self, paint_ctx: &mut PaintCtx, geom: &Geometry) {
+        // todo: Cache resources, and flush cache when the render target is re-created.
+        self.size = geom.size;
+        let resources = self.create_resources(paint_ctx);
+        let rt = paint_ctx.render_target();
+        let rect = RectF::from((0.0, 0.0, self.size.0, self.size.1));
+        rt.fill_rectangle(rect, &resources.bg);
+
+        let first_line = self.y_to_line(0.0);
+        let last_line = min(self.y_to_line(self.size.1) + 1, self.line_cache.height());
+
+        let x0 = LEFT_PAD;
+        let mut y = self.line_to_content_y(first_line) - self.scroll_offset;
+        for line_num in first_line..last_line {
+            if let Some(textline) = self.get_text_line(line_num) {
+                textline.draw_bg(rt, x0, y, &resources.sel);
+            }
+            y += LINE_SPACE;
+        }
+        let mut y = self.line_to_content_y(first_line) - self.scroll_offset;
+        for line_num in first_line..last_line {
+            if let Some(textline) = self.get_text_line(line_num) {
+                textline.draw_text(rt, x0, y, &resources.fg);
+                textline.draw_cursor(rt, x0, y, &resources.fg);
+            }
+            y += LINE_SPACE;
+        }
+        self.resources = Some(resources);
+    }
+
+    fn layout(&mut self, bc: &BoxConstraints, _children: &[Id], _size: Option<(f32, f32)>,
+        _ctx: &mut LayoutCtx) -> LayoutResult
+    {
+        LayoutResult::Size(bc.constrain((0.0, 0.0)))
+    }
+
+    fn mouse(&mut self, event: &MouseEvent, _ctx: &mut HandlerCtx) -> bool { 
+        let MouseEvent { x, y, mods: _, which, count } = *event;
+        if which == MouseButton::Left && count == 1 {
+            let (line, col) = self.xy_to_line_col(x, y);
+            let params = json!({
+                "ty": "point_select",
+                "line": line,
+                "col": col,
+            });
+            self.send_edit_cmd("gesture", &params);
+        }
+        false
+    }
+
+    fn mouse_moved(&mut self, _x: f32, _y: f32, _ctx: &mut HandlerCtx) {}
```

Comment:
Not necessary to impl no-op functions. This one of course will be fleshed out in the future, to handle drag gestures.